### PR TITLE
Dependencies: Switch from openssl to rustls

### PR DESCRIPTION
**Pull request by ecpullen**
**Issue number:**

N/A

**Description of changes:**

Switches from openssl-tls to rustls-tls.

**Testing done:**

Make images.
Tested with `cargo make test`
```
 NAME                                 TYPE        STATE       PASSED   SKIPPED   FAILED
 x86-64-aws-k8s-124                   Resource    completed
 x86-64-aws-k8s-124-instances-xzko    Resource    running
 x86-64-aws-k8s-124-quick             Test        pass        1        6972      0
 ```

**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

### CODE REVIEW

**Review by etungsten — APPROVED** [review]
Awesome!

**Review by mjsterckx — APPROVED** [review]

**Review by jpmcb — COMMENTED** [review]
I see a few references still to `OPENSSL_STATIC` in the example agents: https://github.com/bottlerocket-os/bottlerocket-test-system/blob/53f319a48c3b95dd25f6f9b2c89fb57eafd6ea69/agent/resource-agent/examples/duplicator_resource_agent/Dockerfile#L11

probably makes sense to remove those too?

---

also this phony target to `sdk-openssl` doesn't seem relevant anymore?

https://github.com/bottlerocket-os/bottlerocket-test-system/blob/53f319a48c3b95dd25f6f9b2c89fb57eafd6ea69/Makefile#L34

File: agent/test-agent-cli/examples/example_test_agent_cli/Dockerfile [hunk 1]
```diff
@@ -4,13 +4,9 @@ FROM ${BUILDER_IMAGE} as build
...
 ARG ARCH
 USER root
-# We need these environment variables set for building the `openssl-sys` crate
-ENV PKG_CONFIG_PATH=/${ARCH}-bottlerocket-linux-musl/sys-root/usr/lib/pkgconfig
-ENV PKG_CONFIG_ALLOW_CROSS=1
-ENV CARGO_HOME=/src/.cargo
-ENV OPENSSL_STATIC=true
+
 ENV CARGO_HOME=/src/.cargo
-ADD ./ /src/
```
Comment (jpmcb) [inline agent/test-agent-cli/examples/example_test_agent_cli/Dockerfile:13]: Still need this line for adding the source?

File: agent/test-agent-cli/examples/example_test_agent_cli/Dockerfile [hunk 2]
```diff
@@ -4,13 +4,9 @@ FROM ${BUILDER_IMAGE} as build
 
 ARG ARCH
 USER root
-# We need these environment variables set for building the `openssl-sys` crate
-ENV PKG_CONFIG_PATH=/${ARCH}-bottlerocket-linux-musl/sys-root/usr/lib/pkgconfig
-ENV PKG_CONFIG_ALLOW_CROSS=1
-ENV CARGO_HOME=/src/.cargo
```
Comment (jpmcb) [inline agent/test-agent-cli/examples/example_test_agent_cli/Dockerfile:10]: Still need this line for adding the cargo home?

Comment (ecpullen) [reply agent/test-agent-cli/examples/example_test_agent_cli/Dockerfile:10]: It was there twice.

File: .github/workflows/build.yaml [hunk 3]
```diff
@@ -49,4 +49,4 @@ jobs:
     steps:
       - uses: actions/checkout@v3
       - run: rustup default 1.67.1
-      - run: make cargo-deny
+      - run: make cargo-deny
```
Comment (jpmcb) [inline .github/workflows/build.yaml]: Meant to remove the newline here?

File: deny.toml [hunk 4]
```diff
@@ -19,8 +19,8 @@ allow = [
     "ISC",
     "MIT",
     "OpenSSL",
-    "Unicode-DFS-2016"
-    #"Unlicense",
+    "Unicode-DFS-2016",
+    # "Unlicense",
```
Comment (jpmcb) [inline deny.toml]: Nit: extra space and hanging comma?

**Review by jpmcb — COMMENTED** [review]
nit: could also remove this unused `sdk-openssl` `.PHONY` from the makefile:
https://github.com/bottlerocket-os/bottlerocket-test-system/blob/df0c62d2c5a89b0a7dbfe901a307c4d951fac2b4/Makefile#L34

**Review by jpmcb — APPROVED** [review]

**Review by etungsten — APPROVED** [review]

**Pull request merged**